### PR TITLE
[FIX] account_edi_ubl_cii: wrong journal selected when uploading an invoice in a company with multiple journals of the same type

### PR DESCRIPTION
**Pull request by rigr-odoo**
Steps to reproduce:
- Create two sales journals
- Select the second journal
- Upload an invoice
- The invoice is in the first journal instead of the second

Fix:
We can simply use the context to retrieve the current selected journal if it's given, otherwise we use the previous fallback (infering the journal with the move type)

task-id 3258956

### CODE REVIEW

File: addons/l10n_account_edi_ubl_cii_tests/tests/test_xml_cii_fr.py [hunk 1]
```diff
@@ -310,6 +310,32 @@ def test_import_partner_facturx(self):
...
+            'default_account_id': self.company_data['default_account_revenue'].id,
+            'company_id': self.company_data['company'].id,
+        })
+        invoice = self._generate_move(
+            seller=self.partner_1,
+            buyer=self.partner_2,
+            move_type='out_invoice',
+            invoice_line_ids=[{'product_id': self.product_a.id}],
+        )
+        edi_attachment = invoice.edi_document_ids.filtered(lambda doc: doc.edi_format_id.code == 'facturx_1_0_05').attachment_id.id
```
Comment (william-andre) [inline addons/l10n_account_edi_ubl_cii_tests/tests/test_xml_cii_fr.py:331]:
```suggestion
        edi_attachment = invoice._get_edi_attachment(self.env.ref('account_edi_ubl_cii.edi_facturx_1_0_05')).id
```

File: addons/l10n_account_edi_ubl_cii_tests/tests/test_xml_cii_fr.py [hunk 2]
```diff
@@ -310,6 +310,32 @@ def test_import_partner_facturx(self):
...
+
+        new_invoice = self.env['account.journal'].with_context(default_move_type='out_invoice')._create_document_from_attachment(edi_attachment)
+        self.assertEqual(new_invoice.journal_id, self.company_data['default_journal_sale'])
+
+        new_invoice = self.env['account.journal'].with_context(default_journal_id=journal2.id)._create_document_from_attachment(edi_attachment)
+        self.assertEqual(new_invoice.journal_id, journal2)
```
[the remark this answers is not in the document]
Comment (rigr-odoo) [reply addons/l10n_account_edi_ubl_cii_tests/tests/test_xml_cii_fr.py:337]: For instance `default_journal_id` is set when opening the journal from the Dashboard (here https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_journal_dashboard.py#L523)

File: addons/l10n_account_edi_ubl_cii_tests/tests/test_xml_cii_fr.py [hunk 3]
```diff
@@ -310,6 +310,27 @@ def test_import_partner_facturx(self):
         new_invoice = self._import_invoice_attachment(invoice, 'facturx_1_0_05', self.company_data['default_journal_purchase'])
         self.assertEqual(self.partner_1, new_invoice.partner_id)
 
+    def test_import_journal_facturx(self):
+        """
+        If the context contains the info about the current default journal, we should use it
+        instead of infering the journal from the move type.
+        """
+        journal2 = self.company_data['default_journal_sale'].copy()
+        journal2.default_account_id = self.company_data['default_account_revenue'].id
```
Comment (william-andre) [inline addons/l10n_account_edi_ubl_cii_tests/tests/test_xml_cii_fr.py:319]: Why is this needed? Is the default account not copied?

File: addons/l10n_account_edi_ubl_cii_tests/tests/test_xml_cii_fr.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/l10n_account_edi_ubl_cii_tests/tests/test_xml_cii_fr.py:332]: Mmmh, I guess it's ok; but this test could easily break (by not being useful) since we don't test that the context is indeed there where we need it.

**Review by william-andre — APPROVED** [review]
@robodoo r+
